### ISSUE TITLE
Tensorflow/keras custom loss problem (external inputs)

### ISSUE THREAD

**Issue by paulhigazi** [tensorflow/tensorflow#38790]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- Tensorflow 1.15
- Yes

Lets assume that we have a model model_A and we want to build up a backpropagation based on 3 different loss functions. The first loss (`Loss_1`) should be based on the output of model_A, `Loss_2 `and `Loss_3 `can come from something else. Think about it like a deviation from an unknown source, like in process-automation if you want to build up ur PID-controller. The easiest way is my approach down there, but it actually fails, because the graph isnt constructed the way i want, because X_realB and X_realC have no connection to model_A, and are ignored by keras with tensorflow backend.  

**Main Question:** How can i pass new variables in my loss function, without processing them in the model, and stil influencing the minimization problem ?

```
def generator_model(model_A):

  model_A.trainable = True

# import
  X_realA = Input(shape=image_shape)
  X_realB = Input(shape=image_shape)
  X_realC = Input(shape=image_shape)

# generate Fake image
  Fake_A=model_A(X_realA)


  model = Model([X_realA],[Fake_A,X_realB ,X_realC])

  opt = Adam(lr=0.0002, beta_1=0.5)
  model.compile(loss=["mse","mse","mse"],loss_weights=[1,1,1], optimizer=opt)
  model.summary()
  return model
```

I tried something else in the past 2 days. Wrapping `[FakeA,B,C]` in a custom lambda-layer, to calculate combined loss (one value output of that custom layer). Than passing this loss, in a dummy custom loss-function, which just outputs the combined value of the lambda layer. Here is an example:

```
    # import A,B,C and than pass A into Generator .... and after that:
    
    combined_loss= Lambda(lambda x: combined_loss_func(x))([FakeA,B,C])

    model=Model([A,B,C],[combined_loss],loss=dummy_loss)

    def dummy_loss(y_pred,y_true):
      return y_pred


`combined_loss` could look like that:
    
    def combined_loss_func(x):
    
      FakeA,B,C=x[0],x[1],x[2]
    
      # transform all inputs into one row-tensors
      shape=tf.shape(FakeA)
      FakeA=tf.reshape(FakeA,[1,shape[0]*shape[1]*shape[2]*shape[3]])   
      shape=tf.shape(B)
      B=tf.reshape(B,[1,shape[0]*shape[1]*shape[2]*shape[3]]) 
      shape=tf.shape(C)
      C=tf.reshape(C,[1,shape[0]*shape[1]*shape[2]*shape[3]]) 
    
      # build up a hypothetical ground truth
      FakeA_ones=tf.ones_like(FakeA)
      A_ones=tf.ones_like(A)
      B_ones=tf.ones_like(B)
    
      # calculate losses
      loss0=keras.losses.mse(FakeA,FakeA_ones)
      loss1=keras.losses.mse(A,A_ones)
      loss2=keras.losses.mse(B,B_ones)
    
      # sum them up
      summe=tf.math.add(loss0,loss1)
      summe=tf.math.add(summe,loss2)
    
      # average them
      avg=tf.math.truediv(summe,3.0)
      avg=tf.expand_dims(summe,axis=-1)
    
      return avg
```

If i now try, to set the FakeA loss to zero, no backpropagation to `modelA` happens anymore, or at least nothing in the system changes anymore:

```
       # calculate losses
      loss0=keras.losses.mse(FakeA,FakeA_ones) * 0
      loss1=keras.losses.mse(A,A_ones)
      loss2=keras.losses.mse(B,B_ones)
```

First it seemes really good, but when i go now into the custom-function, and not use `FakeA`, which is the one and only tensor which passed through the generator. Than i stil get a value for my loss function, which seems to be rigth, but actually nothing is happening, my cycle Gan isnt improving at all, and all images passed though stil look the same, even after 100 epochs.

**Comment by paulhigazi** [comment]
Any ideas ?

**Comment by karmel** [comment]
This question is better asked on StackOverflow, where there is a more robust community of users to help. If there is a specific bug you have found, please open a new issue describing the bug.
